builtin/providers/aws/resource_aws_codepipeline.go

```diff
@@ -0,0 +1,504 @@
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"role_arn": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"run_order": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+func validateAwsCodePipelineEncryptionKeyType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"KMS": true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: encryption_key type can only be KMS"))
+	}
+	return
+}
+
+func validateAwsCodePipelineArtifactStoreType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"S3": true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: artifact_store type can only be S3"))
+	}
+	return
+}
+
+func validateAwsCodePipelineStageActionCategory(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"Source":   true,
+		"Build":    true,
+		"Deploy":   true,
+		"Test":     true,
+		"Invoke":   true,
+		"Approval": true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: category can only be one of Source | Build | Deploy | Test | Invoke | Approval"))
+	}
+	return
+}
+
+func validateAwsCodePipelineStageActionOwner(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	types := map[string]bool{
+		"AWS":        true,
+		"ThirdParty": true,
+		"Custom":     true,
+	}
+
+	if !types[value] {
+		errors = append(errors, fmt.Errorf("CodePipeline: owner can only be one of AWS | ThirdParty | Custom"))
+	}
+	return
+}
+
+func validateAwsCodePipelineStageActionConfiguration(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(map[string]interface{})
+	types := map[string]bool{
+		"OAuthToken": true,
+	}
+
+	for k := range value {
+		if types[k] {
+			errors = append(errors, fmt.Errorf("CodePipeline: OAuthToken should be set as environment variable 'GITHUB_TOKEN'"))
+		}
+	}
+	return
+}
+
+func resourceAwsCodePipelineCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codepipelineconn
+	pipeline := expandAwsCodePipeline(d)
+	params := &codepipeline.CreatePipelineInput{
+		Pipeline: pipeline,
+	}
+
+	var resp *codepipeline.CreatePipelineOutput
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		var err error
+
+		resp, err = conn.CreatePipeline(params)
+
+		if err != nil {
+			return resource.RetryableError(err)
+		}
+
+		return resource.NonRetryableError(err)
+	})
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error creating CodePipeline: %s", err)
+	}
+	d.SetId(*resp.Pipeline.Name)
```

Comment:
It does not seem like it, it should raise an error if the pipeline was not created. But I can add a check to be sure.